### PR TITLE
added java transaction processor and 'clear jar' feature

### PR DESCRIPTION
**Pull request by raneja14**
added java transaction processor and 'clear jar' feature

Signed-off-by: raghav.aneja <raghav.aneja@intel.com>

### CODE REVIEW

File: javaprocessor/src/main/java/cookiejar/processor/CookieJarProcessor.java [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+	private void bakeCookies(State state, Integer cookieCount, String userPublicKey, String userAddress)
+			throws InternalError, InvalidTransactionException {
+
+		// Get existing count from ledger state
+		Map<String, ByteString> ledgerMap = state.getState(Collections.singletonList(userAddress));
+		String currStateCount = ledgerMap.get(userAddress).toStringUtf8();
+		logger.info("Baking " + cookieCount + " cookies");
+		// getState() will return empty map if user address key doesn't exist in state
+		if (currStateCount.isEmpty()) {
+			 logger.info("No cookie jar availble");
```
Comment (danintel) [inline javaprocessor/src/main/java/cookiejar/processor/CookieJarProcessor.java:167]: s/availble/available/

File: javaprocessor/src/main/java/cookiejar/processor/CookieJarProcessor.java [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+	@Override
+	public void apply(TpProcessRequest request, State state) throws InvalidTransactionException, InternalError {
+		// Get the user's signing public key from header
+		String userPublicKey = request.getHeader().getSignerPublicKey();
+		// Get the unique user address key derived from the user's signing public key
+		String userAddress = getUserAddress(userPublicKey);
+		logger.info("User Public Key = " + userPublicKey + " ; Address Key = " + userAddress);
+		// Extract the payload as utf8 string from the transaction, in request variable
+		String payload = request.getPayload().toStringUtf8();
+	     // Split the csv utf-8 string
```
Comment (danintel) [inline javaprocessor/src/main/java/cookiejar/processor/CookieJarProcessor.java:89]: Indent this line the same as the other lines.

File: javaprocessor/src/main/java/cookiejar/processor/CookieJarProcessor.java [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+}
+
+
+/* ******************************************************************************
+ * CookieJarHandler
+ *
+ * It handles the processing of operations(bake/eat/clear) supported by cookiejar application.
+ * It sets the name space prefix, versions and transaction family name.
+ * This is the place where you implement your transaction specific logic(Inside apply() method
```
Comment (danintel) [inline javaprocessor/src/main/java/cookiejar/processor/CookieJarProcessor.java:41]: Delete this line (line 41)

File: javaprocessor/src/main/java/cookiejar/processor/CookieJarProcessor.java [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+	}
+
+	@Override
+	public String transactionFamilyName() {
+		return TXN_FAMILY_NAME;
+	}
+
+	private String getUserAddress(String userPublicKey) {
+		// Generate unique key(user address key) from the cookiejar namespace
+        // and user signer public key
```
Comment (danintel) [inline javaprocessor/src/main/java/cookiejar/processor/CookieJarProcessor.java:137]: Indent this line (137) the same as the other lines.

**Comment by raneja14** [comment]
@danintel Please review the changes and merge to your repository

**Review by danintel — CHANGES_REQUESTED** [review]

**Review by danintel — APPROVED** [review]
I'll make the corrections and changes I previously suggested in a follow-up push.

**Pull request merged**